go/vt/vtgate/engine/sequential.go

```diff
@@ -0,0 +1,140 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"context"
+
+	"vitess.io/vitess/go/sqltypes"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/vtgate/evalengine"
+)
+
+// Concatenate Primitive is used to concatenate results from multiple sources.
+var _ Primitive = (*Sequential)(nil)
+
+// Sequential specified the parameter for concatenate primitive
+type Sequential struct {
+	Sources []Primitive
+}
+
+// NewSequential creates a Sequential primitive.
+func NewSequential(Sources []Primitive) *Sequential {
+	return &Sequential{
+		Sources: Sources,
+	}
+}
+
+// RouteType returns a description of the query routing type used by the primitive
+func (s *Sequential) RouteType() string {
+	return "Sequential"
+}
+
+// GetKeyspaceName specifies the Keyspace that this primitive routes to
+func (s *Sequential) GetKeyspaceName() string {
+	res := s.Sources[0].GetKeyspaceName()
+	for i := 1; i < len(s.Sources); i++ {
+		res = formatTwoOptionsNicely(res, s.Sources[i].GetKeyspaceName())
+	}
+	return res
+}
+
+// GetTableName specifies the table that this primitive routes to.
+func (s *Sequential) GetTableName() string {
+	res := s.Sources[0].GetTableName()
+	for i := 1; i < len(s.Sources); i++ {
+		res = formatTwoOptionsNicely(res, s.Sources[i].GetTableName())
+	}
+	return res
+}
+
+// TryExecute performs a non-streaming exec.
+func (s *Sequential) TryExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantFields bool) (*sqltypes.Result, error) {
+	finalRes := &sqltypes.Result{}
+	for _, source := range s.Sources {
+		res, err := source.TryExecute(ctx, vcursor, bindVars, wantFields)
+		if err != nil {
+			return nil, err
+		}
+		finalRes.RowsAffected += res.RowsAffected
+		if finalRes.InsertID == 0 {
+			finalRes.InsertID = res.InsertID
+		}
+		if res.Info != "" {
+			finalRes.Info = res.Info
+		}
+	}
+	return finalRes, nil
+}
+
+// TryStreamExecute performs a streaming exec.
+func (s *Sequential) TryStreamExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantFields bool, callback func(*sqltypes.Result) error) error {
+	for _, source := range s.Sources {
+		err := source.TryStreamExecute(ctx, vcursor, bindVars, wantFields, callback)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetFields fetches the field info.
+func (s *Sequential) GetFields(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable) (*sqltypes.Result, error) {
+	res, err := s.Sources[0].GetFields(ctx, vcursor, bindVars)
```

Comment:
Hmm... Is this really neccessary? As far as I can see, this engine primitive only supports inputs that do not return any rows (`TryExecute` and the streaming version don't return the inputs rows), so there should not be any columns in the output, right? I think this method could simply return an empty `Result` struct

Reply:
marked it as internal bug